aws-frauddetector-detector/src/aws_frauddetector_detector/helpers/update_worker_helpers.py

```diff
@@ -155,16 +183,26 @@ def delete_unused_detector_versions_for_detector_update(afd_client,
         )
 
 
-def delete_unused_rules_for_detector_update(afd_client,
-                                            detector_id: str,
-                                            unused_rule_versions: Set[Tuple[str, str]]):
+def delete_unused_rules_for_detector_update(
+    afd_client, detector_id: str, unused_rule_versions: Set[Tuple[str, str]]
+):
+    # For now, just catch conditional check failed exception, which means the rule is still used.
+    # We will follow up with a more optimal approach (and avoid the try/catch)
     for unused_rule_id, unused_rule_version in unused_rule_versions:
-        api_helpers.call_delete_rule(
-            frauddetector_client=afd_client,
-            detector_id=detector_id,
-            rule_id=unused_rule_id,
-            rule_version=unused_rule_version
-        )
+        try:
+            api_helpers.call_delete_rule(
+                frauddetector_client=afd_client,
+                detector_id=detector_id,
+                rule_id=unused_rule_id,
+                rule_version=unused_rule_version,
+            )
+        except afd_client.exceptions.ConflictException as conflictException:
+            LOG.warning(
+                f"Conflict exception when deleting rule! Continuing without failure. "
+                f"This is likely from a rule being present in an active detector version. "
+                f"This can happen when transitioning from ACTIVE -> DRAFT, and keeping the ACTIVE version. "
+                f"Exception: {conflictException}"
+            )
```

Comment:
We could get a bit smarter here, but this was the fastest path to unblock. Will follow up offline.